.github/workflows/build.yml

```diff
@@ -0,0 +1,122 @@
+
+# Run this workflow every time a push or pull-request on master branch gets submitted
+on:
+  push:
+    branches: [ master ]
+
+  pull_request:
+    branches: [ master ]
+
+  # scheduled for 2AM UTC
+  schedule:
+  - cron: '0 2 * * *'
+
+  workflow_dispatch:
+
+jobs:
+  # Set the job key. The key is displayed as the job name
+  # when a job name is not provided
+  build:
+    name: Build
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Ruby 2.5
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.5
+
+      # Checks out a copy of your repository on the ubuntu-latest machine
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Setup Node
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '10.13.0'
+
+      # Npm global packages caching
+      - name: Cache global npm modules
+        id: global-node-modules-cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-global-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux
+          path: ~/.npm
+          key: ${{ runner.os }}-global-build-cache-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-global-build-cache-${{ hashFiles('**/package-lock.json') }}
+
+      # Node modules caching
+      - name: Cache node modules
+        id: node-modules-cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm modules cache files are stored in `./node_modules`
+          path: ./node_modules
+          key: ${{ runner.os }}-build-cache-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-cache-${{ hashFiles('**/package-lock.json') }}
+
+      # Bundler modules caching
+      - name: Cache bundler gems
+        id: bundler-gems-cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-bundler-gems
+        with:
+          # bundler modules cache files are stored in `./node_modules`
+          path: ./vendor/bundle
+          key: ${{ runner.os }}-bundler-build-cache-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-bundler-build-cache-${{ hashFiles('**/Gemfile.lock') }}
+
+      # Restart mySQL service
+      - name: Before Install
+        run: sudo service mysql restart
+
+      # Install bundler
+      - name: Install bundler
+        run: gem install bundler:1.17.3
+
+      # Run bundler setup
+      - name: Setup Bundler
+        if: steps.bundler-gems-cache.outputs.cache-hit != 'true'
+        run: bundle install --path vendor/bundle
+
+      # Install yarn dependencies
+      - name: Run yarn
+        if: steps.node-modules-cache.outputs.cache-hit != 'true' || steps.global-node-modules-cache.outputs.cache-hit != 'true'
+        run: yarn
+
+      # Run yarn add ember-cli
```

Comment:
I think those kind of commands are redundant :wink: pretty clear what the command does